### ISSUE TITLE
aerender fails on Mac OS 11.4 with error -1701 "AEGetParamPt failed at line 918"

### ISSUE THREAD

**Issue by hdngr** [inlife/nexrender#607]
EDITED

For all who come next.  To be clear, this is a problem with AE Render and the newest OS Security Updates, not nexrender.  My best guess is it is how AE Render uses it's websockets interface to communicate across the machine - kind of squirly and open ended with AE Render sending/receiving requests from the After Effects process.

**Describe the bug**
After upgrading to newest MacOS patches aerender chokes.

Seems to be a permissions issue where aerender cannot open After Effects.

Solution:
1. Open After Effects with any project, or no project at all - just make sure the process is open and running
1. Pass `-reuse` flag to `aerender`:

```
# If using nexrender programatically: 
{
...
aeParams: ['reuse'],
...
}

#Or, if via command line:
>> nexrender [other params] --reuse
```

**Comment by stevenzielke** [comment]
> For all who come next.
> 
> **Describe the bug**
> After upgrading to newest MacOS patches aerender chokes.
> 
> Seems to be a permissions issue where aerender cannot open After Effects.
> 
> Solution:
> Pass `-reuse` flag to `aerender`.
> 
> Using nexrender programatically:
> 
> ```json
> {
> ...
> aeParams: ['reuse'],
> ...
> }
> ```
> 
> Via command line:
> `nexrender ... --reuse`

unfortunately I cannot get the --reuse working as mentioned. You use it like this?
./nexrender-worker --reuse --host=***** --secret=*****

**Comment by torresf** [comment]
@stevenzielke You need to open `After Effects Render Engine` _before_ running your `nexrender-worker`.
Hope this helps !

**Comment by stevenzielke** [comment]
> @stevenzielke You need to open `After Effects Render Engine` _before_ running your `nexrender-worker`.
> Hope this helps !

Thank you so much. That worked. Now I dont have to downgrade my MacOS

**Comment by limzykenneth** [comment]
`reuse` here doesn't work for me as I'm getting `aerender ERROR -1712: AESend failed to send apple event at line 909` instead when setting the reuse flag.

**Comment by hdngr** [comment]
> `reuse` here doesn't work for me as I'm getting `aerender ERROR -1712: AESend failed to send apple event at line 909` instead when setting the reuse flag.

Do you already have AE Open and Running?  Try it with AE Already open and running, and then AE Render will using the existing process and not throw an error.  That is what I and other user's are having success with.

**Comment by limzykenneth** [comment]
Yes, I'm sure I have.

**Comment by kimown** [comment]
https://discussions.apple.com/thread/252832741

https://community.adobe.com/t5/after-effects/aerender-error-1701/td-p/12043974/page/2

There is a same bug in AE&&Mac OSX.

**Comment by chrisapplegate** [comment]
I got an update from Adobe in the [support forum](https://community.adobe.com/t5/after-effects-discussions/aerender-error-1701/m-p/12372908#M180702) - the issue has been fixed in the latest version of After Effects Beta, v22.0.0x94 (note that this is different from the stable version). I've tested it with my own install of nexrender on Catalina, and it works fine as it did before the security update broke things.

So an alternative for now if `reuse=true` is inconvenient or not an option is to install the AE Beta separately. A couple of caveats: first the AE Beta uses its own distinct settings from the stable release so you will need to re-enable "Allow Scripts to Write Files and Access Network" in After Effects preferences if your scripts rely on that. Second, you will need to explicitly set `binary` to the path of the After Effects Beta in the options when initialising nexrender rather than relying on autofind to pick it up.
